pkg/manager/member/tidb_upgrader.go

```diff
@@ -56,6 +56,10 @@ func (tdu *tidbUpgrader) Upgrade(tc *v1alpha1.TidbCluster, oldSet *apps.Stateful
 		return nil
 	}
 
+	if *newSet.Spec.Replicas == int32(0) {
```

Comment:
Need a comment to explain why we should return here, also, I think this predicate can be moved to the beginning of this method